### PR TITLE
Pass python_requires argument to setuptools

### PR DESCRIPTION
**Pull request by jdufresne**
Helps pip decide what version of the library to install.

https://packaging.python.org/tutorials/distributing-packages/#python-requires

> If your project only runs on certain Python versions, setting the
> python_requires argument to the appropriate PEP 440 version specifier
> string will prevent pip from installing the project on other Python
> versions.

https://setuptools.readthedocs.io/en/latest/setuptools.html#new-and-changed-setup-keywords

> python_requires
>
> A string corresponding to a version specifier (as defined in PEP 440)
> for the Python version, used to specify the Requires-Python defined in
> PEP 345.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
     author="Prasanth Nair",
     author_email="prasanthhn@gmail.com",
     url='https://github.com/phn/jdcal',
+    python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*",
```
Comment (phn) [inline setup.py]: Hello Jon,

Does this mean that pip won't install on 3.0.*, 3.1.*, 3.2.* and 3.3.*?

If so, isn't this a dangerous thing to do? 

As far as I know the library works on all versions of Python >=2.6. In addition the audience for this library is all software that depends on openpyxl and related libraries[1]. In fact, because of this dependency jdcal is among the top downloaded libraries on pypi!! 

Perhaps the versions of Python you have listed are not used much and won't create too much problems, but we never know. And if not making this change helps even just a handful of people we should not make the change, given the limited benefits of the change.

So, in short, I am not in favor of making this change for the above reasons. At the same time, thanks for your contribution and hope to receive more help from you in the future.

I will wait for your comments before I close this PR.

[1]: https://github.com/phn/jdcal/issues/5

Regards,
Prasanth

Comment (jdufresne) [reply setup.py]: For any released package that includes the `python_requires`, yes your understanding is correct. When pip installs jdcal, it will only consider versions that are supported by the running Python. Past jdcal releases will continue to install. You can try this with a package like [Pillow](https://github.com/python-pillow/Pillow). If you install in a Python 3.6 environment, pip installs the latest version. If you install in a Python 3.3 environment, an older version will install.

I can adjust the version string to support the version range of the package. What versions of Python are supported?

The benefits of the change are that the package will not be installed in a unsupported environment. This is especially useful when it is known the package does not work in older environments. This allows jdcal to evolve overtime in a way that is more compatible with existing Python environments as new releases will only install on supported environments.

As jdcal is only a single Python file, and if it supports all version of Python, perhaps it is not needed.

I followed the versions the package supports by examining the trove classifiers. Are these inaccurate? If so, perhaps the trove classifier should be updated as well. Looking at `.travis.yml` I see Python 2.6 is included (but it is not in `tox.ini`). I can also update these files to include all supported versions as well.

To add some data to mix, using [pypinfo](https://github.com/ofek/pypinfo), here are the download statistics for the past 30 days:

```
$ pypinfo --pip --percent --markdown jdcal pyversion
```

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 2.7            |  58.19% |        141,571 |
| 3.6            |  20.85% |         50,730 |
| 3.4            |  11.87% |         28,869 |
| 3.5            |   8.82% |         21,454 |
| 2.6            |   0.12% |            293 |
| 3.7            |   0.11% |            273 |
| 3.3            |   0.04% |             96 |
| None           |   0.00% |              5 |
| 3.2            |   0.00% |              1 |

If you still prefer not to merge this, no worries at all.

Comment (phn) [reply setup.py]: Hello Jon,

Thank you for the detailed comments. I didn't know about pypinfo.

1) You are correct in that the trove classifiers are incorrect. The package should work on all Python >=2.6. I will update the classifier and add these in travis.yml as well.

2) As you can see from the pypinfo. There are people installing jdcal from all Python >=2.6 except 3.0,  and 3.1 has only 1 download.  I do not envisage making any significant changes to the code base and so there is not much of an evolution to speak of, other than adjusting for incompatible changes for some future Python.

So my conclusion is that if we do add ``python_requires`` it will have to have all  Python versions >=2.6 and not exclude any. Does that serve the purpose of ``python_requires``? 

Regards,
Prasanth

Comment (jdufresne) [reply setup.py]: I agree with your analysis. At that point, `python_requires` doesn't really serve any purpose. I think it is best to close this.

Thanks!
